Wangscape/tilegen/alpha/CalculatorTopTwo.cpp

```diff
@@ -0,0 +1,54 @@
+#include "CalculatorTopTwo.h"
+
+#include <algorithm>
+#include <cassert>
+
+namespace tilegen
+{
+namespace alpha
+{
+
+
+void CalculatorTopTwo::updateAlphasAux(const Weights& weights)
+{
+    initialiseIndexedWeights(weights);
+    std::sort(mIndexedWeights.begin(), mIndexedWeights.end(), std::greater<IndexedWeight>());
+    const IndexedWeight& winner = mIndexedWeights[0];
+    const IndexedWeight& runner_up = mIndexedWeights[1];
+    const IndexedWeight& base = mIndexedWeights[2];
+
+    double gap_1 = winner.first - runner_up.first;
+    double gap_2 = runner_up.first - base.first;
```

Comment:
`gap_2` is used in `double runner_up_scaled = pow(gap_2 / total_gap, power);`. `gap_1` could be removed, if `assert(total_gap >= gap_2)` was added.